artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQMessage.java

```diff
@@ -423,7 +421,7 @@ public Destination getJMSDestination() throws JMSException {
          SimpleString address = message.getAddressSimpleString();
          SimpleString name = address;
 
-         if (address != null & enable1xPrefixes) {
+         if (address != null & session.isEnable1xPrefixes()) {
```

Comment:
Hot path here

Reply:
I don't see a difference from the previous code. 

This is getJMSDestination. Not really on the hot path.

Reply:
It can be heavily called in apps doing dynamic re routing.

Reply:
Why does session it need to be changable?

Reply:
Session is not changeable. 

the property on session currently is changeable because of the ResourceAdapter. 

We can verify if it can be final there even for RA.

Reply:
notice the previous version had a boolean per message. I'm improving this by using nly one at the session.

Reply:
I got that this is better but tbh the commit that added thay change was merged fairly rapidly too. And once we relesse we get stuck with it. So rather we tidy up as much as possible now.

Reply:
@michaelandrepearce  you made a comment I can't find on the web interface here. could only see on email.

```
Eg the jms connection or jms session
```

The Resource Adapter will not have reference to the JMS Session in the same way. CoreSession is the only way to do it.

The other possibility would be to use a metadata, but I did not want to trust a hashmap operation everytime this was called.